pkg/fqdn/dnsproxy/proxy.go

```diff
@@ -522,7 +530,14 @@ func (p *DNSProxy) CheckAllowed(endpointID uint64, destPort uint16, destID ident
 //  - Write the response to the endpoint.
 func (p *DNSProxy) ServeDNS(w dns.ResponseWriter, request *dns.Msg) {
 	stat := ProxyRequestContext{}
+	stat.SemaphoreAcquireTime.Start()
+	if p.ConcurrencyLimit != nil {
+		p.ConcurrencyLimit.Acquire(context.TODO(), 1)
+		defer p.ConcurrencyLimit.Release(1)
+	}
+	stat.SemaphoreAcquireTime.End(true)
```

Comment:
Should the semaphore acquire timing only be executed when concurrency is being limited?

```suggestion
	if p.ConcurrencyLimit != nil {
		stat.SemaphoreAcquireTime.Start()
		p.ConcurrencyLimit.Acquire(context.TODO(), 1)
		defer p.ConcurrencyLimit.Release(1)
		stat.SemaphoreAcquireTime.End(true)
	}
```